### ISSUE TITLE
How to use SAM2 if my own project already initialized Hydra?

### ISSUE THREAD

**Issue by royvelich** [facebookresearch/sam2#179]
Hi,
In `https://github.com/facebookresearch/segment-anything-2/blob/main/sam2/__init__.py`, you initialize Hydra by calling `initialize_config_module.` But what if my application has already initialized Hydra for its purposes as well? In this case, Hydra will be already initialized, and the following exception will be raised:
`ValueError: GlobalHydra is already initialized, call GlobalHydra.instance().clear() if you want to re-initialize`.

In other words, if Hydra is already initialized, then the following import `from sam2.sam2_video_predictor import SAM2VideoPredictor` will crash the execution.

Does anyone know how to solve this? Maybe, instead of calling `initialize_config_module`, you should call Hydra's `compose`?

Thanks

**Comment by royvelich** [comment]
IMHO, before executing the line `initialize_config_module("sam2_configs", version_base="1.2")`, there's a need to attempt to check whether Hydra is already initialized. If so, the end-user should ensure that the required YAML config files are located in his own project's `config` directory. There should also be an option to let the user provide the relative path to the folder in which the config files are located (relative to his project's main Hydra config folder).

**Comment by YeonhwaCha** [comment]
Is there a way to solve this problem right now?

**Comment by royvelich** [comment]
> Is there a way to solve this problem right now?

See my pull request

**Comment by GuHuangAI** [comment]
@royvelich 
Hello, I use your code, but it is still wrong.
I found the reason is that `GlobalHydra().is_initialized()` returned `False`, and my project would call `initialize_config_module`.
ValueError: GlobalHydra is already initialized, call GlobalHydra.instance().clear() if you want to re-initialize.

**Comment by haithamkhedr** [comment]
This should be resolved [here](https://github.com/facebookresearch/sam2/blob/52198ead0eb13ae8270bea6ca768ef175f5bf167/sam2/__init__.py#L10) if you pull the latest main.

**Comment by kishanml** [comment]
```
# %cd segment-anything-2
import hydra
from sam2.build_sam import build_sam2
from sam2.sam2_image_predictor import SAM2ImagePredictor

sam2_checkpoint = "path"
model_cfg = "path"

sam2_model = build_sam2(model_cfg, sam2_checkpoint, device="cuda")
predictor = SAM2ImagePredictor(sam2_model)

predictor.model.sam_mask_decoder.train(True)
predictor.model.sam_prompt_encoder.train(True)

```


AssertionError: GlobalHydra is not initialized, use @hydra.main() or call one of the hydra initialization methods first

Can someone help me with this ?

**Comment by JamesMcCullochDickens** [comment]
> ```
> # %cd segment-anything-2
> import hydra
> from sam2.build_sam import build_sam2
> from sam2.sam2_image_predictor import SAM2ImagePredictor
> 
> sam2_checkpoint = "path"
> model_cfg = "path"
> 
> sam2_model = build_sam2(model_cfg, sam2_checkpoint, device="cuda")
> predictor = SAM2ImagePredictor(sam2_model)
> 
> predictor.model.sam_mask_decoder.train(True)
> predictor.model.sam_prompt_encoder.train(True)
> ```
> 
> AssertionError: GlobalHydra is not initialized, use @hydra.main() or call one of the hydra initialization methods first
> 
> Can someone help me with this ?

Did you end up finding a solution?

**Comment by danielkorth** [comment]
Having the same issue. Any ideas how to fix it?
